### ISSUE TITLE
Question: How could I get this to update based on a changing value?

### ISSUE THREAD

**Issue by rchrdnsh** [simeydotme/svelte-range-slider-pips#13]
So, I know you mentioned that you had not intended for this to be a timeline/seekbar, etc...but everything else about it is so nice I was wondering if you had any idea as to how I might go about adding that kind of functionality?

My thought is that the incoming value is a reactive value that is changing (seconds from the current time of a piece of music) so the slider would need the ability to update it's value, and therefore it's position, based on the incoming value, and also be able to set a value be being clicked on, which I think it already does?

Anyway, if you have any thoughts you can share i would be most appreciative :-)

**Comment by simeydotme** [comment]
Hi @rchrdnsh  ... 

so firstly; I don't see any reason why you can't/shouldn't use this, but it's not the intended functionality and I doubt there's (m)any components built for time-scrubbing as it's not a particularly common UI feature. Please feel free to fork this range-slider and remove the unnecessary features (ranges/handles/animation/styling/keyboard/etc) and add the ones you need and re-package it as a time-scrubber component :) ~~ however I'm not going to be modifying the functionality to suit your specific needs, or to extend it for use as a different UI component. I'm not super familiar with time-scrubbers so I also wouldn't be confident myself of building a great one!

---

secondly; for sharing Open Source Etiquette ~~ [it's not really great to be shotgunning a ton of questions in to an "bug/issue tracking" repository for code](https://medium.com/@methane/why-you-must-not-ask-questions-on-github-issues-51d741d83fde). While I do really appreciate the attention and that you're enjoying the component... as you can imagine if everyone who uses it asks questions in here it's going to be overwhelming and noisy. Fortunately there's places where asking this kind of question is much better, both for you, and for the community! 

- I'd suggest if you're having problems with a component or implementation then you should ask in StackOverflow first with clear and specific labels so experts in those labels will find and help you. This also means you will gain notoriety and the question will be visible when others are looking for answers in Google. https://stackoverflow.com/questions/tagged/svelte

- If you're having general svelte questions (and not really problems, but perhaps you want advice like in this issue here) then you could join the Svelte discord! There's hundreds (thousands?) of good Svelte developers in there who are generally smarter and more experienced that me, haha. https://discord.gg/GSCFJ2pz

**Comment by rchrdnsh** [comment]
hmmmm, my apologies, I didn't realize I was 'shotgunning' questions, I was just trying to figure out if-and-how I could use this component for my needs and got super into it :-) I'm relatively new to programming (a couple years) and still don't fully understand all of the ecosystem and tooling around it (much less the programming part itself). I'm much more knowledgeable in design and then html/css, etc...

I'm also actually trying to do exactly what you just said, by copying the code over into a REPL and removing features while keeping it working (failing at the moment):

https://svelte.dev/repl/8ed9a33bfec14e68a2c742d96e563f42?version=3.29.6

...and I did not expect you to add those features, simply if you had any idea about how they might be implemented, based on your experience. Again, my apologies if it came across that way.

I have personally had too many bad experiences on stack overflow, so I'm reluctant to use it, but the svelte discord has been great so far. I do get replies like: "Can you ask the component author?"...which is what led me to ask questions here. It would be nice, in my opinion, if Github had separate categories/tabs for issues and questions, as I agree that conflating the two is not good, which is generally why I try to close out questions as soon as I can to keep the issues clean.

So, to reiterate, it was not my intention to pollute the repo with questions, just excited by your work and really wanting to bend it to my needs :-)

**Comment by simeydotme** [comment]
> I have personally had too many bad experiences on stack overflow, so I'm reluctant to use it, but the svelte discord has been great so far. I do get replies like: "Can you ask the component author?"...which is what led me to ask questions here. It would be nice, in my opinion, if Github had separate categories/tabs for issues and questions, as I agree that conflating the two is not good, which is generally why I try to close out questions as soon as I can to keep the issues clean.

^ Totally agree on this, and I get about StackOverflow; it's really hit and miss. I've seen it deteriorate over time as new developers have an ever increasing chasm of knowledge to bridge and more experienced Devs probably feel like they are repeating themselves a lot. And thanks for considering the issue tracker being clean! 🙇  I don't consider your questions as pollution, haha, just that a lot came at once. But I totally get you're excited and sorry I am not able to facilitate everything.

But that article I linked has some good points, and especially something it missed is that I sign up for notifications to my repos as I want to do a good job of patching anything or catching any problems which are critical, but it's a double-edge blade when the notifications are simply questions. It also leads me (or us as maintainers) feeling sometimes guilty and dejected by not being able to respond to, or help, every question. I'm lucky right now as I only maintain a few things and it's a quiet moment where I can give you some attention! :)

------

> I'm also actually trying to do exactly what you just said, by copying the code over into a REPL and removing features while keeping it working (failing at the moment)

^ this is a good idea, and you've made good progress... but the REPL is quite difficult to debug and you effectively have double the code in there to debug as the two components have the same internals. You might have a more fruitful time if you download the repo and use the `/test/` folder while editing with the `yarn run dev` for quick debugging :)

hint: `getClosestHandle()` should just return `0` if you're not using ranges, or just bypass it altogether by saving a single reference to the handle with `bind:this={handle}` (check Svelte docs)

Good luck mate and sorry if I discouraged you at all!
